### ISSUE TITLE
'All the way down' crashes on linux even with the plugins built

### ISSUE THREAD

**Issue by i30817** [adventuregamestudio/ags#790]
The plugins it needs are `libagslua.so` and `libags-spritefont-plugin.so`. They're on the main repo, so just copying a build of them by looking for the .so's  in the Plugin dir allows the game to try to start.

However, it then crashes with 
```

Adventure Game Studio v3.5 Interpreter
Copyright (c) 1999-2011 Chris Jones and 2011-2019 others
ACI version 3.5.0.12

Initializing allegro
Initializing game data
Located game data file: /media/i30817/Huggin/Downloads/games/AllTheWayDownV1/deepuns3.exe
Opened game data file: game28.dta
Game data version: 44
Compiled with: 3.3.0.1167
Setting up game configuration
Logging scipt to "warnings.log"
Data directory: /media/i30817/Huggin/Downloads/games/AllTheWayDownV1/
Voice pack found and initialized.
audio.vox found and initialized.
Setting up window
Initializing TTF renderer
Initializing keyboard
Initializing mouse: number of buttons reported is 3
Sound settings: digital driver ID: 'AUTO' (0xffffffff), MIDI driver ID: 'AUTO' (0xffffffff)
Trying to init: digital driver ID: 'SDL2' (0x53444c32), MIDI driver ID: 'DIGI' (0x44494749)
Installed digital driver ID: 'SDL2' (0x53444c32), MIDI driver ID: 'DIGI' (0x44494749)
Install exit handler
Initialize path finder library
Game GUI version: 117
Plugin 'ags-spritefont-plugin' loaded as 'libags-spritefont-plugin.so', resolving imports...
Plugin 'agslua' loaded as 'libagslua.so', resolving imports...
[PLUGIN] AGSSpriteFont: Init fixed width renderer
[PLUGIN] AGSSpriteFont: Init vari width renderer
[PLUGIN] AGSSpriteFont: Register functions
An internal error has occurred. Please note down the following information.
If the problem persists, post the details on the AGS Technical Forum.
(ACI version 3.5.0.12)

Error: [Lua] attempt to call a string value

***** ENGINE HAS SHUTDOWN
```

Not a big deal if it's _supposed_ to be incompatible, but just checking.

**Comment by ivan-mogilko** [comment]
lua plugin is severely outdated, which may be a reason, or one of them.

**Comment by i30817** [comment]
The short freeware game 'Salt' by the makers of primordia has the same error.

Here is the game: https://plnehry.idnes.cz/salt-recenze-c93-/Download.aspx?id=D131027_996267

**Comment by ivan-mogilko** [comment]
Running same version of the engine but with original windows agslua.dll from the game package actually works. So I guess there's an inconsistency in lua plugin. Perhaps the sources are either newer or older. I recall we had continious trouble with lua plugin which had numerous updates and some did not work correctly.

**Comment by i30817** [comment]
all the way down has (according to exiftool) 

a original `lua5.1.dll` of version:
```
File Version Number             : 5.1.4.0
Product Version Number          : 5.1.4.0
File Version                    : 5.1.4
```

and a agslua.dll doesn't have a version (ofc).

diff proves both dlls are the same on both games.

**Comment by i30817** [comment]
['Barely Floating' is another game with a lua dependency that breaks.](https://www.adventuregamestudio.co.uk/site/games/game/1597/)

**Comment by rofl0r** [comment]
> Error: [Lua] attempt to call a string value

sounds like a debugging session is due

**Comment by ivan-mogilko** [comment]
I mentioned this above, lua plugin was in development and had several versions. I believe one should begin with a research to find out which version is currently referenced in our repository (it is linked as submodule), and which versions are used in these games, if plugin is backwards compatible, and so on.

**Comment by i30817** [comment]
Wouldn't it be easier to try to adapt the glue plugin to work with the current release of lua5.1? Were people using this plugin actually coding lua scripts/code or just calling bindings?

**Comment by i30817** [comment]
The fact that error occurs in all games at startup makes me think it's between the plugin and the dll, but that raises the question of if the game devs could send strings as lua code as part of the plugin interface if they wouldn't screw up in the same way.

https://stackoverflow.com/questions/4036743/scite-lua-string-comparison-raises-attempt-to-call-a-string-value

Has a likely culprit on the comments: `!=` confused for `~=` 

or maybe one of these:
https://www.lua.org/manual/5.1/manual.html#7
or these:
https://www.lua.org/manual/5.0/manual.html
(search for "Incompatibilities with version 4.0")

**Comment by ivan-mogilko** [comment]
The original game in "All the way down" uses lua 5.1.

**Comment by i30817** [comment]
> > Error: [Lua] attempt to call a string value
> 
> sounds like a debugging session is due

GDB is not helping on this case, even on a debug build - 
```
Error: [Lua] attempt to call a string value

[Thread 0x7fffeffff700 (LWP 12815) exited]
[Thread 0x7ffff59da700 (LWP 12814) exited]
***** ENGINE HAS SHUTDOWN
[Thread 0x7ffff43ea700 (LWP 12816) exited]
[Inferior 1 (process 12805) exited with code 0133]
(gdb) bt
No stack.

```

**Comment by rofl0r** [comment]
> GDB is not helping on this case, even on a debug build -

yeah, i meant looking for where the error comes from an setting a bp there

**Comment by i30817** [comment]
Wouldn't you have to build lua51.so for that? Anyway, i'm not that knowledgeable about debugging to do that, my experience is limited to 'build debug, wait for crash, backtrace - so i fear I'll be of little to no help.

**Comment by rofl0r** [comment]
a start could be to modify the lua source code to instead of displaying mentioned error message use abort(); (or even better `__asm__("int3");` - that would cause getting a backtrace in gdb, and then the memory contents can be examined
